agent/src/test/java/com/teamscale/jacoco/agent/store/upload/delay/DelayedCommitDescriptorStoreTest.java

```diff
@@ -1,56 +1,74 @@
 package com.teamscale.jacoco.agent.store.upload.delay;
 
 import com.teamscale.client.CommitDescriptor;
-import com.teamscale.jacoco.agent.util.InMemoryStore;
+import com.teamscale.jacoco.agent.TestBase;
+import com.teamscale.jacoco.agent.util.InMemoryUploader;
 import org.junit.jupiter.api.Test;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class DelayedCommitDescriptorStoreTest {
+public class DelayedCommitDescriptorStoreTest extends TestBase {
 
 	@Test
-	public void shouldStoreToCacheIfCommitIsNotKnown() {
-		InMemoryStore cache = new InMemoryStore();
-		InMemoryStore destination = new InMemoryStore();
-		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, cache);
+	public void shouldStoreToCacheIfCommitIsNotKnown() throws IOException {
+		File coverageFile = File.createTempFile("jacoco-", ".xml");
+		Path outputPath = coverageFile.getParentFile().toPath();
 
-		store.store("xml1");
+		InMemoryUploader destination = new InMemoryUploader();
+		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, outputPath);
 
-		assertThat(cache.getXmls()).containsExactly("xml1");
-		assertThat(destination.getXmls()).isEmpty();
+		store.upload(coverageFile);
+
+		assertThat(
+				Files.list(outputPath).anyMatch(path -> path.getFileName().equals(Paths.get(coverageFile.getName()))))
```

Comment:
if you're comparing file names, I'd prefer `path.getFileName().toString().equals(coverageFile.getName())` that is easier to understand for me